### DISCUSSION TITLE
Spatie Activity Log

### DISCUSSION THREAD

**Discussion by ComputerMaverick69** [spatie/laravel-activitylog#1134] [Q&A]
Is there a way to include customization of `ActivityLog`

I'm looking for a way to included the events and properties affected in the description such that it shows specifics. For example `User A updated their email address to abc@xyz.com`. Considering that `properties keeps track of the changes`. 

Also, is there a way to isolate the events fired? Like say, if a user creates and account.

I currently have the below code in my user model but i want for more customisations
```
public function getActivitylogOptions(): LogOptions { 
        return LogOptions::defaults()
                        ->logOnly([ 'name', 'age', 'fitness_level', 'fitness_location', 'fitness_goal', 'height', 'weight', 'fitness_frequency', 'status', 'subscription'])
                        ->logOnlyDirty()
                        ->dontSubmitEmptyLogs()
                        ->setDescriptionForEvent(fn(string $eventName) => auth()->user()->name." {$eventName} "); 
    }
```
I also know to use the below but it requires me to programmatically write the logs. I was looking for something more dynamic.

```
activity()
   ->performedOn($anEloquentModel)
   ->causedBy($user)
   ->withProperties(['customProperty' => 'customValue'])
   ->log('Look, I logged something');
```

**Comment by deonthomasgy** [comment]
I am looking for a similar solution, did you manage to find a solution that works for you?

**Reply by ComputerMaverick69** [reply]
No. I had to settle to programmatically log events... It's time consuming though and incomplete still.

**Reply by deonthomasgy** [reply]
this sucks. By chance would you know how to get the old-attributes and new ones in the getActivityLogOptions. I am in the process of writing a func to get most of this done dynamically
